### PR TITLE
Added black box switcher test

### PR DESCRIPTION
**Pull request by sd1ver**

### CODE REVIEW

File: synapse-grid-core/src/main/scala/ru/primetalk/synapse/core/dsl/Switcher2Dsl.scala [hunk 1]
```diff
@@ -20,6 +20,7 @@ trait Switcher2Dsl extends SystemBuilderDsl{
         {
           case Signal(_, s@Signal(contact, _)) if outputContacts.contains(contact) =>
             Iterable.single(s)
+          case other => throw new IllegalStateException(s"Unexpected signal $other")
```
Comment (Primetalk) [inline synapse-grid-core/src/main/scala/ru/primetalk/synapse/core/dsl/Switcher2Dsl.scala]: In fact we can be more specific about error. We know that it's a `Signal` in data. And we can just say that `contact` of input signal is not in `outputContacts` list.

File: synapse-grid-core/src/main/scala/ru/primetalk/synapse/core/package.scala [hunk 2]
```diff
@@ -47,3 +47,4 @@ package object core
   with SystemConvertingApi
   with BaseTypedSystemApi
   with SwitcherDsl
+  with Switcher2Dsl
```
Comment (Primetalk) [inline synapse-grid-core/src/main/scala/ru/primetalk/synapse/core/package.scala]: :+1:

File: build.sbt [hunk 3]
```diff
@@ -46,6 +46,11 @@ lazy val akka = (project in file("synapse-grid-akka")).settings(
   libraryDependencies += "com.typesafe.akka" %% "akka-slf4j" % akkaVersion
 ).dependsOn(slf4j)
 
+lazy val examples = (project in file("synapse-grid-examples")).settings(
```
Comment (Primetalk) [inline build.sbt]: :+1:

File: synapse-grid-core/src/main/scala/ru/primetalk/synapse/core/runtime/SignalProcessingApi0.scala [hunk 4]
```diff
@@ -26,7 +26,7 @@ trait SignalProcessingApi0 extends SignalsApi with TrellisApi with RuntimeCompon
     type TrellisElement = (Context, TSignals)
     type TrellisElementTracking = (Context, TSignals)
     type TrellisProducerTracking = TotalTrellisBuilder => TSignals => TSignals
-    type TotalTrellisProducerTracking = ((Context, Signal[_]) => TrellisElementTracking)
+    type TotalTrellisProducerTracking = (Context, Signal[_]) => TrellisElementTracking
```
Comment (Primetalk) [inline synapse-grid-core/src/main/scala/ru/primetalk/synapse/core/runtime/SignalProcessingApi0.scala]: yep

File: synapse-grid-core/src/main/scala/ru/primetalk/synapse/core/runtime/SignalProcessingApi0.scala [hunk 5]
```diff
@@ -119,6 +119,11 @@ trait SignalProcessingApi0 extends SignalsApi with TrellisApi with RuntimeCompon
                 val (ns, signals) = f.asInstanceOf[(Any, Signal[_]) => (Any, List[Signal[_]])](s, signal)
                 trellisBuilder.currentState = trellisBuilder.currentState.asInstanceOf[Map[Contact[Any], Any]].updated(sh, ns).asInstanceOf[Context]
                 trellisBuilder.addSignals(trace, proc, signals)
+              case BlackBoxRuntimeComponent(_, _, _, f) =>
```
Comment (Primetalk) [inline synapse-grid-core/src/main/scala/ru/primetalk/synapse/core/runtime/SignalProcessingApi0.scala]: I thought that if it's `RuntimeComponent` there should be a branch that supports that interface. If not, we could replace this with
```
case rc: RuntimeComponent =>
```

File: synapse-grid-core/src/test/scala/ru/primetalk/synapse/core/BlackBoxSwitcherSpec.scala [hunk 6]
```diff
@@ -0,0 +1,35 @@
+package ru.primetalk.synapse.core
+
+import org.scalatest.{FlatSpecLike, Matchers}
+
+class BlackBoxSwitcherSpec extends FlatSpecLike with Matchers {
```
Comment (Primetalk) [inline synapse-grid-core/src/test/scala/ru/primetalk/synapse/core/BlackBoxSwitcherSpec.scala]: Very nice

**Review by Primetalk — APPROVED** [review]
LGTM.
With a couple of comments.

File: synapse-grid-core/src/main/scala/ru/primetalk/synapse/core/dsl/Switcher2Dsl.scala (hunk 1 above)
Comment (sd1ver) [reply synapse-grid-core/src/main/scala/ru/primetalk/synapse/core/dsl/Switcher2Dsl.scala]: done

File: synapse-grid-core/src/main/scala/ru/primetalk/synapse/core/runtime/SignalProcessingApi0.scala (hunk 5 above)
Comment (sd1ver) [reply synapse-grid-core/src/main/scala/ru/primetalk/synapse/core/runtime/SignalProcessingApi0.scala]: Done.

**Pull request merged**